jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/handler/SendSignalAction.java

```diff
@@ -0,0 +1,81 @@
+import org.jbpm.workflow.core.node.Transformation;
+import org.kie.api.runtime.manager.RuntimeManager;
+import org.kie.api.runtime.process.ProcessContext;
+
+public class SendSignalAction implements Action {
+
+    private String signalName;
+    private String varName;
+    private boolean isAsync;
+    private NodeImpl node;
+
+    public SendSignalAction(NodeImpl node, String variable, String signalName, boolean isAsync) {
+        this.node = node;
+        this.varName = variable;
+        this.signalName = signalName;
+        this.isAsync = isAsync;
+    }
+
+    @Override
+    public void execute(ProcessContext kcontext) throws Exception {
+        Object tVariable = kcontext.getVariable(varName);
+        Transformation transformation = (Transformation) kcontext.getNodeInstance().getNode()
+                .getMetaData().get("Transformation");
+        if (transformation != null) {
+            tVariable = new EventTransformerImpl(transformation).transformEvent(tVariable);
+        }
+        String signalType = VariableUtil.resolveVariable(isAsync ? "ASYNC-" + signalName : signalName, kcontext
+                .getNodeInstance());
+        String scope = (String) node.getMetaData("customScope");
+        if ("processInstance".equalsIgnoreCase(scope)) {
+            kcontext.getProcessInstance().signalEvent(signalType, tVariable);
+        } else if ("runtimeManager".equalsIgnoreCase(scope) || "project".equalsIgnoreCase(scope)) {
+            ((RuntimeManager) kcontext.getKieRuntime().getEnvironment().get("RuntimeManager")).signalEvent(signalType,
+                    tVariable);
+        } else if ("external".equalsIgnoreCase(scope)) {
+            ((InternalProcessRuntime) kcontext.getKieRuntime()).getProcessEventSupport().fireOnSignal(kcontext
+                    .getProcessInstance(), kcontext.getNodeInstance(), kcontext.getKieRuntime(), signalName, tVariable);
+            WorkItemImpl workItem = new WorkItemImpl();
+            workItem.setName("External Send Task");
+            workItem.setNodeInstanceId(kcontext.getNodeInstance().getId());
+            workItem.setProcessInstanceId(kcontext.getProcessInstance().getId());
+            workItem.setNodeId(kcontext.getNodeInstance().getNodeId());
+            workItem.setDeploymentId((String) kcontext.getKieRuntime().getEnvironment().get("deploymentId"));
+            workItem.setParameter("Signal", signalType);
+            workItem.setParameter("SignalProcessInstanceId", kcontext.getVariable("SignalProcessInstanceId"));
+            workItem.setParameter("SignalWorkItemId", kcontext.getVariable("SignalWorkItemId"));
+            workItem.setParameter("SignalDeploymentId", kcontext.getVariable("SignalDeploymentId"));
+            if (tVariable != null) {
+                workItem.setParameter("Data", tVariable);
+            }
+            ((WorkItemManager) kcontext.getKieRuntime().getWorkItemManager()).internalExecuteWorkItem(workItem);
```

Comment:
Should we catch WorkItemHandlerNotFoundException in case user has not registered a handler for signal